app/services/reinstate_declined_offer.rb

```diff
@@ -0,0 +1,16 @@
+class ReinstateDeclinedOffer
+  def initialize(course_choice:, zendesk_ticket:)
+    @course_choice = course_choice
+    @zendesk_ticket = zendesk_ticket
+  end
+
+  def save!
+    @course_choice.update!(
+      status: 'offer',
+      declined_at: nil,
+      declined_by_default: false,
```

Comment:
Question, i actually don't know. Should we allow this service to be used on DBD's? Does that fit under accidental decline? If not then we need to this line 

`      rows = if FeatureFlag.active?(:support_user_reinstate_offer) && application_choice.declined?
`

so it's 

`      rows = if FeatureFlag.active?(:support_user_reinstate_offer) && application_choice.declined? && !application_choice.declined_by_default
`

Reply:
Changed & update spec to include a check for this

Reply:
I think we decided not to support DBD for this PR?

Reply:
Yea this can come out then, since you can't access the page to reinstate an offer for a dbd application choice.